dev-setup.sh

```
@@ -0,0 +1,290 @@
+    fail "Node.js is required."
+  fi
+  ok "Node.js installed ($(node --version))"
+fi
+
+# ── Rust / Cargo ──────────────────────────────────────────────────────────────
+if command -v cargo &>/dev/null; then
+  skip "Rust/Cargo ($(cargo --version))"
+else
+  info "Installing Rust via rustup..."
+  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+  source "$HOME/.cargo/env"
```

Shellcheck (0.11.0)

[info] Not following: ./.cargo/env was not specified as input (see shellcheck -x).

(SC1091)